packages/vsce/__tests__/__e2e__/util/initSetup.test.ts

```diff
@@ -122,48 +135,119 @@ export async function getRegionResourceIndex(regionResources: TreeItem[], resour
   return -1;
 }
 
-export async function getResourceInRegion(
+export async function expectTreeItemIsSelected(treeItem: TreeItem | undefined) {
+  const isSelected = await treeItem?.getAttribute("aria-selected");
+  expect(isSelected).to.equal("true");
+}
+
+export async function verifyProgramAttributes(programName: string, expectedAttributes: { [key: string]: string }) {
+  const editorView = new EditorView();
+
+  const webView = (await editorView.openEditor(`CICSProgram PROGLIB(${programName})`)) as WebView;
+  expect(webView).not.undefined;
+  await webView.switchToFrame();
+
+  const table = await webView.findWebElement(By.id("resultsTable"));
+  expect(table).not.undefined;
+
+  const tableRows = await table.findElements(By.css("tr"));
+  expect(tableRows).not.empty;
+
+  const searchBar = await webView.findWebElement(By.id("searchBox"));
+  expect(searchBar).not.undefined;
+
+  for (const [attributeKey, expectedValue] of Object.entries(expectedAttributes)) {
+    await searchBar.clear();
+    await searchBar.sendKeys(attributeKey);
+    await searchBar.sendKeys("\uE007"); // Enter key
+
+    const visibleRows: WebElement[] = [];
+    for (const row of tableRows) {
+      if (await row.isDisplayed()) {
+        visibleRows.push(row);
+      }
+    }
+
+    let found = false;
+    for (const row of visibleRows) {
+      const keyElement = await row.findElement(By.css("th"));
+      const key = (await keyElement.getText()).trim().toLowerCase();
+
+      if (key === attributeKey.toLowerCase()) {
+        const value = await row.findElement(By.css("td")).getText();
+        expect(value).contains(expectedValue);
+        found = true;
+        break;
+      }
+    }
+
+    expect(found, `Attribute ${attributeKey} not found`).to.be.true;
+  }
+
+  await searchBar.clear();
+  await webView.switchBack();
+}
+
+export async function getLabelAfterArrowDown(driver: WebDriver): Promise<string> {
+  await driver.actions().sendKeys(Key.ARROW_DOWN).perform();
+  const activeElement: WebElement = await driver.switchTo().activeElement();
+  const fullText: string = await activeElement.getText();
+  const label: string = fullText.split("\n")[0].trim();
+  return label;
+}
+
+export async function setupCICSTreeSelectedResourceParams(
   cicsTree: DefaultTreeSection,
   profileName: string,
   plexName: string,
   regionName: string,
   resourceName: string
-): Promise<TreeItem[]> {
-  const plexChildren = await getPlexChildren(cicsTree, profileName, plexName);
-  expect(plexChildren).not.empty;
+) {
+  // Get children of the plex
+  const cicsex61Children = await getPlexChildren(cicsTree, profileName, plexName);
```

Comment:
I reckon this variable name is misleading - it could be children of any plex that's passed in right? Not hardcoded to one of our mocked sample plexes?